pkg/ingester/instance.go

```diff
@@ -1144,3 +1146,22 @@ func minTs(stream *logproto.Stream) model.Time {
 	}
 	return model.TimeFromUnixNano(streamMinTs)
 }
+
+// For each stream, we check if the stream is owned by the ingester or not and increment/decrement the owned stream count.
+func (i *instance) updateOwnedStreams(ownedTokenRange ring.TokenRanges) error {
+	i.ownedStreamsSvc.resetStreamCounts()
+
+	err := i.forAllStreams(context.Background(), func(s *stream) error {
+		if ownedTokenRange.IncludesKey(uint32(s.fp)) {
+			i.ownedStreamsSvc.incOwnedStreamCount()
+		} else {
+			i.ownedStreamsSvc.incNotOwnedStreamCount()
+		}
+		return nil
+	})
```

Comment:
we must add a lock on the streams, because between `resetStreamCounts` and `forAllStreams` other goroutines can create a stream and already increase the counter, so when we iterate over all streams, we will increment the counter again for the streams that are created in the middle...

Reply:
done